libs/src/services/ethContracts/index.js

```diff
@@ -273,8 +273,15 @@ class EthContracts {
       var highestFoundSPVersion = foundVersionsList.sort(semver.rcompare)[0]
       // Randomly select from highest found endpoints
       let highestFoundSPVersionEndpoints = spVersionToEndpoint[highestFoundSPVersion]
-      var randomValidSPEndpoint = highestFoundSPVersionEndpoints[Math.floor(Math.random() * highestFoundSPVersionEndpoints.length)]
-      selectedServiceProvider = randomValidSPEndpoint
+      
+      // Sort by duration ASC so quickest responding discovery providers come first
+      highestFoundSPVersionEndpoints.sort((a, b) => {
```

Comment:
Probably want to eliminate that randomly select comment above - will hold off looking further until marked for review